src/app/components/HighlightedText/text-cutting.ts

```diff
@@ -0,0 +1,74 @@
+import { findTextMatch, NormalizerOptions } from './text-matching'
+
+export interface CutAroundOptions extends NormalizerOptions {
+  /**
+   * What should be the length of the fragment delivered, which
+   * has the pattern inside it?
+   *
+   * The default value is 80.
+   */
+  fragmentLength?: number
+}
+
+/**
+ * Return a part of the corpus that contains the match to the pattern, if any
+ *
+ * If the corpus is undefined or empty, undefined is returned.
+ *
+ * If either the pattern is undefined or empty, or there is no match,
+ * an adequately sized part from the beginning of the corpus is returned.
+ *
+ * If there is a match, but the corpus is at most as long as the desired fragment length,
+ * the whole corpus is returned.
+ *
+ * If there is a match, and the corpus is longer than the desired fragment length,
+ * then a part of a corpus is returned, so that the match is within the returned part,
+ * around the middle.
+ */
+export function cutAroundMatch(
+  corpus: string | undefined,
+  pattern: string | undefined,
+  options: CutAroundOptions = {},
+): {
+  hasMatch: boolean
+  part: string | undefined
+} {
+  const { fragmentLength = 80, ...matchOptions } = options
+
+  if (!corpus) {
+    // there is nothing to see here
+    return {
+      hasMatch: false,
+      part: undefined,
+    }
+  }
+
+  // do we have a match?
+  const match = pattern ? findTextMatch(corpus, [pattern], matchOptions) : undefined
```

Comment:
searching by capitalized pattern is broken (based on the example in PR description):
- https://ec1a5160.oasis-explorer.pages.dev/mainnet/consensus/proposal/3?voter=bam
  ![image](https://github.com/oasisprotocol/explorer/assets/3758846/d923379d-aff3-4062-8a4b-3165e0a5d5fe)
- https://ec1a5160.oasis-explorer.pages.dev/mainnet/consensus/proposal/3?voter=Bam
  ![image](https://github.com/oasisprotocol/explorer/assets/3758846/8d9468f6-7d7f-4a68-8a62-a560a1b5e59b)

Reply:
Good catch.

Here, the problem was that the referenced example PR didn't use the correct routine (`findTextMatch()`) to look for the match. (It _was_ using the highlighting functionality from here.)

Fixed this on the other PR by calling the correct function.